### ISSUE TITLE
W3F Revisions

### ISSUE THREAD

**Issue by 0x4007** [ubiquity/partnerships#30]
> Thanks for the application @pavlovcik a couple of initial comments/questions:
> 
> - Regarding the milestones, the mandatory deliverables look good (0a. - 0d.) but delivery table descriptions 1 - 6 seem to still be the template > boilerplate text. Can you replace this with the concrete technical deliverables that you would produce? 
> - We only fund software that doesn't exist yet, and it seems that some of the [DevPool](https://dao.ubq.fi/the-devpool) tools have already been > built, such as [ubiquibot](https://github.com/ubiquity/ubiquibot). Can you specify in the application what you would be building as part of the > grant? The more technical details the better.
> - For any front-end focused milestones, we typically ask for mockups to be presented in the application.
> - Please also note that we typically only fund ink! smart contracts, as we aren't Solidity experts.

_Originally posted by @keeganquigley in https://github.com/w3f/Grants-Program/pull/1800#pullrequestreview-1478244982_

**Comment by 0x4007** [comment]
@sergfeldman @Hodlatoor

**Comment by sergfeldman** [comment]
> > Thanks for the application @pavlovcik a couple of initial comments/questions:
> > 
> > * Regarding the milestones, the mandatory deliverables look good (0a. - 0d.) but delivery table descriptions 1 - 6 seem to still be the template > boilerplate text. Can you replace this with the concrete technical deliverables that you would produce?
> > * We only fund software that doesn't exist yet, and it seems that some of the [DevPool](https://dao.ubq.fi/the-devpool) tools have already been > built, such as [ubiquibot](https://github.com/ubiquity/ubiquibot). Can you specify in the application what you would be building as part of the > grant? The more technical details the better.
> > * For any front-end focused milestones, we typically ask for mockups to be presented in the application.
> > * Please also note that we typically only fund ink! smart contracts, as we aren't Solidity experts.
> 
> _Originally posted by @keeganquigley in [w3f/Grants-Program#1800 (review)](https://github.com/w3f/Grants-Program/pull/1800#pullrequestreview-1478244982)_

W3F requests technical deliverables on ther chains, but we do not plan to create and provide them.

**Comment by 0x4007** [comment]
Is our answer that we will not continue? I feel like there's probably a way to pull this off while still conforming to their requirements.

**Comment by Hodlatoor** [comment]
@sergfeldman Hi Ser, so we have 4 items to address:

1. Regarding the milestones, the mandatory deliverables look good (0a. - 0d.) but delivery table descriptions 1 - 6 seem to still be the template > boilerplate text. Can you replace this with the concrete technical deliverables that you would produce?

What concrete deliverables do we want to introduce in your mind? I think we should narrow the scope quite a bit.

2. We only fund software that doesn't exist yet, and it seems that some of the [DevPool](https://dao.ubq.fi/the-devpool) tools have already been > built, such as [ubiquibot](https://github.com/ubiquity/ubiquibot). Can you specify in the application what you would be building as part of the > grant? The more technical details the better.

How about we make this granular? To make things clear, let's call it "UbiquiTask" what it does is what we define in #1. This way it's brand new, clear and the functionality which currently doesn't exist, is easily understood. 

3. For any front-end focused milestones, we typically ask for mockups to be presented in the application.

Let's finalize UbiquiTask and can do SC / Wireframes

4. Please also note that we typically only fund ink! smart contracts, as we aren't Solidity experts.

Doesn't sound like a problem. Ink! projects are on GitHub. We are project agnostic so implicitly we also support Ink! products.


Let's get #1 sorted and the rest falls into place.

**Comment by 0x4007** [comment]
Whoever is opening the pull request and providing the deliverable should be the assignee. Also I'm not keen on a flippant rebrand across the codebase/github, twitter, telegram and docs on the website just for a grant.

**Comment by Hodlatoor** [comment]
So to clarify, Think of it as a wheel. DP is the wheel. The wheel has different components like a tire and a rim. In the scope of this grant and for clarity, what we are building now is the rim. We are not rebranding anything. 

The wheel is DP. The rim is Ubiquitask. Alongside Ubiquibot (the tire) and other code it makes the wheel.

Just naming a specific set of new components we are building under this grant for the sake of clarity.

**Comment by sergfeldman** [comment]
@pavlovcik @Hodlatoor If you decide that it is needed, I can try to add this ^ information.

---

At the moment, I suggest asking W3F in the PR comment like this:

"DevPool is a chain-agnostic solution without technical deliverables on Substrate.
At the same time, DevPool accelerates Open Source development and brings value for the development community.
The details are in the grant application.

We would like to get your feedback on the solution, and if you see the perspectives, then a grant.
Can you evaluate this a bit non-usual grant application?"

**Comment by Hodlatoor** [comment]
I don't think that's a good idea. I think we give them what they ask for.

**Comment by sergfeldman** [comment]
> @sergfeldman Hi Ser, so we have 4 items to address:
> 
> 1. Regarding the milestones, the mandatory deliverables look good (0a. - 0d.) but delivery table descriptions 1 - 6 seem to still be the template > boilerplate text. Can you replace this with the concrete technical deliverables that you would produce?
> 
> What concrete deliverables do we want to introduce in your mind? I think we should narrow the scope quite a bit.
> 
> 2. We only fund software that doesn't exist yet, and it seems that some of the [DevPool](https://dao.ubq.fi/the-devpool) tools have already been > built, such as [ubiquibot](https://github.com/ubiquity/ubiquibot). Can you specify in the application what you would be building as part of the > grant? The more technical details the better.
> 
> How about we make this granular? To make things clear, let's call it "UbiquiTask" what it does is what we define in #1. This way it's brand new, clear and the functionality which currently doesn't exist, is easily understood.
> 
> 3. For any front-end focused milestones, we typically ask for mockups to be presented in the application.
> 
> Let's finalize UbiquiTask and can do SC / Wireframes
> 
> 4. Please also note that we typically only fund ink! smart contracts, as we aren't Solidity experts.
> 
> Doesn't sound like a problem. Ink! projects are on GitHub. We are project agnostic so implicitly we also support Ink! products.
> 
> Let's get #1 sorted and the rest falls into place.


@pavlovcik please clarify, do you approve these 4 activities?

**Comment by 0x4007** [comment]
> @pavlovcik please clarify, do you approve these 4 activities?

I would rather delegate the strategy/decisions to you guys. Besides, the penalty of getting this wrong is basically just opening a new pull request. 

> > 1. Regarding the milestones, the mandatory deliverables look good (0a. - 0d.) but delivery table descriptions 1 - 6 seem to still be the template > boilerplate text. Can you replace this with the concrete technical deliverables that you would produce?

> > What concrete deliverables do we want to introduce in your mind? I think we should narrow the scope quite a bit.

It is most likely going to be milestones on our development roadmap for the bot. 

> > 2. We only fund software that doesn't exist yet, and it seems that some of the [DevPool](https://dao.ubq.fi/the-devpool) tools have already been > built, such as [ubiquibot](https://github.com/ubiquity/ubiquibot). Can you specify in the application what you would be building as part of the > grant? The more technical details the better.

Milestones, again.

> > 3. For any front-end focused milestones, we typically ask for mockups to be presented in the application.

This is generic advice that isn't applicable to our proposal.

> > 4. Please also note that we typically only fund ink! smart contracts, as we aren't Solidity experts.

This is also generic advice that isn't applicable to our proposal. However Hodlatoor's comment is worth adding in.

**Comment by Hodlatoor** [comment]
@sergfeldman let's run some ideas for #1  - keeping it simple and effective.

**Comment by Hodlatoor** [comment]
@pavlovcik do you want to msg noc2 to reopen saying forthcoming?
https://github.com/w3f/Grants-Program/pull/1800#pullrequestreview-1478244982
@sergfeldman let's try and get the deliverables for 1 sorted out so I can modify around it and we can resubmit.

**Comment by 0x4007** [comment]
No need to bother them with another action until we have something to show. Otherwise we are just wasting their time.

**Comment by Hodlatoor** [comment]
@pavlovcik @sergfeldman what do you think if we made the new deliverable something like:
Building the InkHunterBot which will have 3 core components, for eg:

1. Creating and deploying the functionality that allows automated confirmation of completed issues by cross-referencing Github code repositories  and submissions. 
2. Confirmation of successful deliverable by issuer and payment distribution to hunter.
3. Integration with UbiquiBot and specific prioritization for Ink! protocols with 10% bounty bonus for Ink! bounties for the first 6 months.

**Comment by 0x4007** [comment]
1. That's up to the contributors to associate their changes to a bounty. Bad idea to try and automate when the bounty hunters 100% know what problem they are trying to solve vs the bot trying to guess. 
2. We already do that
3. This is just a (manual) incentive model for a specific partner, it isn't building a feature.

**Comment by sergfeldman** [comment]
@pavlovcik @Hodlatoor 
We discussed that for the W3F application Milestones we will use modules or features that doesn't exist yet and are pending or imminent in the DevPool development.

I've reviewed the DevPool roadmap and all open issues.
The modules or features could be related to areas: 
1) Financial Incentives.
2) Artificial Intelligence.

Features related to Financial Incentives look highly relevant because they accelerate Open Source development that could be valuable for W3F.
I have not yet figured out a definite NEW milestone with clear technical deliverables because implementation of many features is in progress now.

Modules or features related to AI can be relevant as well but I'm not sure we will implement them in the near future.

I will continue defining Milestones tomorrow.

**Comment by Hodlatoor** [comment]
@sergfeldman financial incentives should be good. as long as we delineate milestones.

**Comment by sergfeldman** [comment]
@Hodlatoor Please, review the example of the Milestone table.


| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| **0a.** | License | MIT |
| **0b.** | Documentation | We will provide inline documentation |
| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
| **0d.** | GitHub Issue | We will provide GitHub Issues as proof of functionality (this is functional as a deliverable because the bot runs inside of GitHub). We already built the capability for the bot to run automatically off of a GitHub Actions runner bundled within the source code repository, which can be cross-referenced with the GitHub Issues as proof if needed. https://github.com/ubiquity/ubiquibot/issues/311, https://github.com/ubiquity/devpool-directory-bounties/issues/4, https://github.com/ubiquity/devpool-directory/issues/502, https://github.com/ubiquity/ubiquibot/issues/228, https://github.com/ubiquity/ubiquibot/issues/394, https://github.com/ubiquity/ubiquibot/issues/425, https://github.com/ubiquity/ubiquibot/issues/268, https://github.com/ubiquity/ubiquibot/issues/96, https://github.com/ubiquity/ubiquibot/issues/302 |
| 0e. | Article | We will publish an article that explains what was done as part of the grant. Completion of the tasks related to the Open Source Contributor Incentive Suite - a system of financial incentives, notifications, and reports that fairly reward Open Source project contributors. |
| 1. | Substrate module: X | No Substrate deliverables |
| 2. | Substrate module: Y | No Substrate deliverables |
| 3. | Substrate module: Z | No Substrate deliverables |
| 4. | Substrate chain | No Substrate deliverables |
| 5. | Library: ABC | No Substrate deliverables |
| 6. | Smart contracts: ... | No Substrate deliverables |

**Comment by Hodlatoor** [comment]
are you suggesting the work scope will be:

- ubiquity/ubiquibot#311 
- ubiquity/devpool-directory-bounties#4 
- ubiquity/devpool-directory#502 
- ubiquity/ubiquibot#228 
- ubiquity/ubiquibot#394 
- ubiquity/ubiquibot#425 
- ubiquity/ubiquibot#268 
- ubiquity/ubiquibot#96 
- ubiquity/ubiquibot#302

**Comment by 0x4007** [comment]
> fully covered by comprehensive unit tests

Side note but this will be a major project too.

**Comment by Hodlatoor** [comment]
correct me if I'm missing something but it looks like a list of items that are not very jointed and I can see them having a hard time understanding what it is that they're paying for.  
that list doesn't answer a simple question:
"with your funds we will build a ____"

**Comment by Hodlatoor** [comment]
What about a bounty value estimator based on existing bounty data, user data and the betting component? 
Call it: Bounty Expert AI v1.

**Comment by 0x4007** [comment]
> What about a bounty value estimator based on existing bounty data, user data and the betting component? 
> Call it: Bounty Expert AI v1. 

We're already working on crowdsourcing that information with the Thales/Synthetix grant.

**Comment by Hodlatoor** [comment]
> > What about a bounty value estimator based on existing bounty data, user data and the betting component? 
> > Call it: Bounty Expert AI v1. 
> 
> We're already working on crowdsourcing that information with the Thales/Synthetix grant. 

Yes that's why I mentioned it because we can piggyback off of that And I can see that as a module that is defined and purpose-built.

**Comment by 0x4007** [comment]
I expect that crowdsourcing this information will solve the business problem with higher accuracy than estimates over the short-to-mid-term. Perhaps on the scale of years will the purely machine estimation engine be more accurate, but it would make sense to seed its intelligence with the crowdsourced human input first. Afterwards, we can source all of the bounty time estimate intelligence from our partners. 

Either way, at this stage it doesn't make sense to build.

**Comment by Hodlatoor** [comment]
So the problem is they are hesitant to fund existing upgrades. I think we should make this into it's own core component which can be utilized in other applications, DevPool included. To that end, let's call it the Aries Crowd Source Engine for DevPool (or any other name you want) and we build it, alongside hooks and elements which help customize it into DevPool. 

As such, Aries CrowdSource Engine integration into DevPool is comprised of the following unique components and tools:

Milestone 3 (Crowdsource pricing engine logic)

milestone 2 (Aries related process optimization) :
https://github.com/ubiquity/ubiquibot/issues/311
https://github.com/ubiquity/devpool-directory-bounties/issues/4
https://github.com/ubiquity/devpool-directory/issues/502
https://github.com/ubiquity/ubiquibot/issues/268
https://github.com/ubiquity/ubiquibot/issues/302

milestone 1 (Aries related UI/UX enhancements):
https://github.com/ubiquity/ubiquibot/issues/228
https://github.com/ubiquity/ubiquibot/issues/394
https://github.com/ubiquity/ubiquibot/issues/425
https://github.com/ubiquity/ubiquibot/issues/96

**Comment by sergfeldman** [comment]
> @Hodlatoor Please, review the example of the Milestone table.
> 
> Number	Deliverable	Specification
> **0a.**	License	MIT
> **0b.**	Documentation	We will provide inline documentation
> **0c.**	Testing and Testing Guide	Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests.
> **0d.**	GitHub Issue	We will provide GitHub Issues as proof of functionality (this is functional as a deliverable because the bot runs inside of GitHub). We already built the capability for the bot to run automatically off of a GitHub Actions runner bundled within the source code repository, which can be cross-referenced with the GitHub Issues as proof if needed. [ubiquity/ubiquibot#311](https://github.com/ubiquity/ubiquibot/issues/311), [ubiquity/devpool-directory-bounties#4](https://github.com/ubiquity/devpool-directory-bounties/issues/4), [ubiquity/devpool-directory#502](https://github.com/ubiquity/devpool-directory/issues/502), [ubiquity/ubiquibot#228](https://github.com/ubiquity/ubiquibot/issues/228), [ubiquity/ubiquibot#394](https://github.com/ubiquity/ubiquibot/issues/394), [ubiquity/ubiquibot#425](https://github.com/ubiquity/ubiquibot/issues/425), [ubiquity/ubiquibot#268](https://github.com/ubiquity/ubiquibot/issues/268), [ubiquity/ubiquibot#96](https://github.com/ubiquity/ubiquibot/issues/96), [ubiquity/ubiquibot#302](https://github.com/ubiquity/ubiquibot/issues/302)
> 0e.	Article	We will publish an article that explains what was done as part of the grant. Completion of the tasks related to the Open Source Contributor Incentive Suite - a system of financial incentives, notifications, and reports that fairly reward Open Source project contributors.
> 1.	Substrate module: X	No Substrate deliverables
> 2.	Substrate module: Y	No Substrate deliverables
> 3.	Substrate module: Z	No Substrate deliverables
> 4.	Substrate chain	No Substrate deliverables
> 5.	Library: ABC	No Substrate deliverables
> 6.	Smart contracts: ...	No Substrate deliverables

@Hodlatoor The main area for review - questions 1-6 are still without answers. 
Other content is secondary. 

As it is hard to find a new DevPool module that meets all the requirements, I artificially compiled "Open Source Contributor Incentives Suite - a system of financial incentives, notifications, and reports that fairly reward Open Source project contributors" with these ^ currently open issues as examples.

If you prefer an artificially compiled "Aries Crowd Source Engine for DevPool which can be utilized in other applications, DevPool included", it is totally fine for me.

**Comment by Hodlatoor** [comment]
@pavlovcik @sergfeldman Guys - take a look.
https://www.notion.so/ubiquitydao/Aries-Crowd-Source-Engine-extensible-and-deployed-on-DevPool-ef867161bf314e9785315160c9311f70?pvs=4

**Comment by 0x4007** [comment]
We aren't using notion anymore you should be doing this in GitHub in a markdown document added to the repo, and request us as reviewers. 

https://dao.ubq.fi/non-code-devpool-deliverables-process

**Comment by Hodlatoor** [comment]
https://github.com/Hodlatoor/partnerships/blob/2496c4f926c81ceff6b1e05ed1c89c724b5b82a6/Aries%20CS%20Engine

tried to go back and edit to associate the task but couldn't see where. was worried more about the formatting.

also how do I move it into the /github folder and add chores / associate task now that it's there?

**Comment by sergfeldman** [comment]
> https://github.com/Hodlatoor/partnerships/blob/2496c4f926c81ceff6b1e05ed1c89c724b5b82a6/Aries%20CS%20Engine
> 
> tried to go back and edit to associate the task but couldn't see where. was worried more about the formatting.
> 
> also how do I move it into the /github folder and add chores / associate task now that it's there?

@Hodlatoor I created the PR with 3 Milestone tables required for the W3F grant. The content in these Milestone tables is based on yours https://github.com/Hodlatoor/partnerships/pull/1

@pavlovcik Please, let me know if I created the PR not correctly.

**Comment by 0x4007** [comment]
> https://github.com/Hodlatoor/partnerships/blob/2496c4f926c81ceff6b1e05ed1c89c724b5b82a6/Aries%20CS%20Engine
> 
> 
> 
> tried to go back and edit to associate the task but couldn't see where. was worried more about the formatting.
> 
> 
> 
> also how do I move it into the /github folder and add chores / associate task now that it's there?

That's a good start. 

- You need to be sure to append the file extension, "`.md`" as the file name suffix so that GitHub understands and renders it as a markdown document. 
- for directories you can write a "/" in the file name and GitHub will understand it as a directory. An example of this is in my video. 
- finally, if you want to just toss in files into the repository without associating it as a task or go through a review, you can (presumably with the process you did now, which was "add a file" and committing directly to `main`, or `development` on this repository- whichever is the default branch) however what you should do is fork the repo, and toss it into your copy, and then open a pull request (this is in my video) a pull request will allow you to associate it with an issue, and will allow us to review it.  
  - regarding process optimization, you might be able to skip forking and instead while committing a file using the GitHub UI I noticed that there's an option to commit on another branch on this repository. This will still allow you to open a pull request without forking. But forking should be the standard process because anybody can do that, whereas the alternative flow is only contributors added to the organization (typically only internal folks)

**Comment by Hodlatoor** [comment]
@sergfeldman are we all good and submitted here so we can close this task?

**Comment by 0x4007** [comment]
Can you try opening up a pull request and associating this issue with your deliverable?

**Comment by 0x4007** [comment]
lol @0xcodercrane can you tame this bot

**Comment by 0x4007** [comment]
This is actually hilarious to me

**Comment by Hodlatoor** [comment]
since i got the file up there maybe we should just close it?

**Comment by 0x4007** [comment]
Generally the pull request reviews must be completed before the associated issues are marked as complete. Besides, its a good opportunity for you to practice applying changes from a review, such as renaming the file extension.

**Comment by 0x4007** [comment]
The review has been neglected. You should finish the requested changes to close this issue out.
